java/impl/aws/src/main/java/co/worklytics/psoxy/aws/LambdaRequest.java

```diff
@@ -0,0 +1,53 @@
+package co.worklytics.psoxy.aws;
+
+import co.worklytics.psoxy.gateway.HttpEventRequest;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@NoArgsConstructor //for jackson
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Getter
+public class LambdaRequest implements HttpEventRequest {
+
+    String httpMethod;
+
+    String path;
+
+    @JsonProperty("multiValueQueryStringParameters")
+    Map<String, List<String>> queryParameters;
+
+    Map<String, String> pathParameters;
+
+    Map<String, String> stageVariables;
+
+    @JsonProperty("multiValueHeaders")
+    Map<String, List<String>> headers;
+
+    String body;
+
+    @Override
+    public Optional<String> getQuery() {
+        if (this.getQueryParameters() == null) {
+            return Optional.empty();
+        } else {
+            String value = this.getQueryParameters().entrySet().stream()
+                .flatMap(parameter -> parameter.getValue().stream().map(v -> parameter.getKey() + "=" + v))
```

Comment:
does this work as expected for multi-valued query parameters? Is there a test to cover this?

Reply:
Probably safer to use `URIBuilder` and then extract the query string

Reply:
👍  yes, requires adding explicit dep on apache `http-client`, but it's a transitive dep of Google http-client, so OK